### PR TITLE
feat(sources): added us census source

### PR DESCRIPTION
**Pull request by chase-adams**
## Description

This PR adds the US Census as a data source to Chainlink. This includes both the 2010 Decennial Census and the 2013-2019 American Community Survey (5-Year).

A future PR will be made for the 2020 Decennial Census and ACS, once those are released by the US Census Bureau.

## Changes

- Added US Census source.
- Added https://github.com/uscensusbureau/citysdk as a dependency (published by the US Census Bureau).
- Added integration, e2e, and unit tests.

## Quality Assurance

- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

File: packages/core/legos/package.json [hunk 1]
```diff
@@ -137,6 +137,7 @@
     "@chainlink/uniswap-v2-adapter": "1.0.0",
     "@chainlink/uniswap-v3-adapter": "1.0.0",
     "@chainlink/upvest-adapter": "1.0.0",
+    "@chainlink/us-census-adapter": "*",
```
Comment (boxhock) [inline packages/core/legos/package.json:140]: Thanks for making this PR! We recently made some changes here where we pin the version, would you mind doing this here as well?

Comment (chase-adams) [reply packages/core/legos/package.json]: Just updated this! Thanks

**Comment by chase-adams** [comment]
@boxhock just updated these tests to set the API key correctly :)

File: packages/sources/us-census/test/integration/census.test.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        .post('/')
+        .send(data)
+        .set('Accept', '*/*')
+        .set('Content-Type', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(400)
+      expect(response.body).toMatchSnapshot()
+    })
+  })
+})
```
Comment (boxhock) [inline packages/sources/us-census/test/integration/census.test.ts:94]: It looks like all of these test cases are testing for errors. Could we add a test that tests for the correct expected result on a successful run?

Comment (boxhock) [reply packages/sources/us-census/test/integration/census.test.ts:94]: (The returned data can be mocked data for the purpose of this testing, we just want to make sure we have a test that covers the happy path)

File: packages/sources/us-census/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+| :-------: | :-----: | :---------------------------------------------------------------------: | :-----: | :---------: |
+|    ✅     | API_KEY | [US Census API key](https://api.census.gov/data/key_signup.html) to use |         |             |
+
+---
+
+### Input Parameters
+
+| Required? |   Name    |                                              Description                                              |                                        Options                                         | Defaults to |
+| :-------: | :-------: | :---------------------------------------------------------------------------------------------------: | :------------------------------------------------------------------------------------: | :---------: |
+|    ✅     | endpoint  |        The [Census Source/year](https://www.census.gov/data/developers/data-sets.html) to use         | dec_2010, acs5_2013, acs5_2014, acs5_2015, acs5_2016, acs5_2017, acs5_2018, acs5_2019  |             |
```
Comment (boxhock) [inline packages/sources/us-census/README.md:19]: While this makes sense in this context alone, I think this is slightly confusing with what we normally use `endpoint` for in our EA monorepo.

I'd recommend making `endpoint` optional and default to `census` (similar to the format we follow with the other EAs). Then what's currently used as `endpoint` here should rather be called `dataset` or something like that.

Does that make sense to you?

**Review by boxhock — CHANGES_REQUESTED** [review]
PR looks good for the most part! Just left two comments below

File: packages/sources/us-census/README.md (hunk 3 above)
Comment (chase-adams) [reply packages/sources/us-census/README.md:19]: Yeah, that makes a lot of sense. I went back and forth on this, looking at other impls. I'll include that in my changes!

**Review by boxhock — CHANGES_REQUESTED** [review]
Had a few more comments. Should be good to merge once these are resolved.

File: packages/sources/ap-election/test/e2e/election.test.ts [hunk 4]
```diff
@@ -25,7 +25,7 @@ describe('execute', () => {
 
     requests.forEach((req) => {
       it(`${req.name}`, async () => {
-        const adapterResponse = await execute(req.testData as AdapterRequest)
+        const adapterResponse = await execute(req.testData as AdapterRequest, undefined)
```
[the remark this answers is not in the document]
Comment (chase-adams) [reply packages/sources/ap-election/test/e2e/election.test.ts]: Just made this change

File: packages/sources/us-census/test/integration/census.test.ts [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+  //     const response = await req
+  //       .post('/')
+  //       .send(data)
+  //       .set('Accept', '*/*')
+  //       .set('Content-Type', 'application/json')
+  //       .expect('Content-Type', /json/)
+  //       .expect(400)
+  //     expect(response.body).toMatchSnapshot()
+  //   })
+  // })
```
[the remark this answers is not in the document]
Comment (chase-adams) [reply packages/sources/us-census/test/integration/census.test.ts]: Good catch - just uncommented them

File: packages/sources/us-census/src/endpoint/census.ts [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        ...firstResult,
+        result: encodedResult,
+      },
+      result: encodedResult,
+    }
+
+    return Requester.success(jobRunID, respData, config.verbose)
+  } catch (e) {
+    throw Requester.errored(jobRunID, e, 400)
+  }
```
[the remark this answers is not in the document]
Comment (chase-adams) [reply packages/sources/us-census/src/endpoint/census.ts]: Removed this

**Comment by chase-adams** [comment]
Thanks @boxhock - let me know if you see anything else

**Comment by boxhock** [comment]
@chase-adams Looks like there's some yarn/dependency updates needed, mind having a look at that?

**Review by cds95 — APPROVED** [review]

**Comment by boxhock** [comment]
@chase-adams mind checking out the failing integration tests? Seems to also be outputting this quite a lot: `Getting Census statistics data from source`

**Comment by chase-adams** [comment]
> @chase-adams mind checking out the failing integration tests? Seems to also be outputting this quite a lot: `Getting Census statistics data from source`

Yeah, unfortunately that's a byproduct of the `citysdk` dependency. Let me take a look - I may need to remove that dependency due to open handles after the tests are completed. So it might take a bit longer to get this PR to land!

**Comment by austinborn** [comment]
@chase-adams Hello, just wanted to check in and see whether you're still developing this PR? No worries if it's not ready yet, just trying to close out stale PRs.

**Comment by chase-adams** [comment]
Apologies for the delay, I've been traveling - I will get these changes over the weekend! If you'd prefer, I can open another PR if you'd like to clean this one up.

**Comment by austinborn** [comment]
> Apologies for the delay, I've been traveling - I will get these changes over the weekend! If you'd prefer, I can open another PR if you'd like to clean this one up.

No problem, it's up to you if you'd like to open a fresh PR with the relevant changes. Feel free to work on it as necessary and request review from `smartcontractkit/solutions-engineering` when it's ready. If you have any questions or issues let us know, and thank you for contributing!

**Comment by boxhock** [comment]
Hey @chase-adams, it seems like this PR has gone stale. Just to keep things organized I will close this PR, but feel free to re-open and request a review once it's ready!

**Pull request closed without merging**